web/widgets.py

```diff
@@ -24,3 +25,36 @@ def __init__(self, *args, **kwargs):
 
 class SemanticDateInput(forms.DateInput):
     template_name = "web/forms/widgets/semantic_date.html"
+
+
+class MazemapSearchInput(forms.TextInput):
+    """
+    Widget that enables mazemap search functionality, including autofill of url to mazemap
+    """
+    template_name = "web/forms/widgets/mazemap_search.html"
+
+    class Media:
+        js = ('web/js/mazemap-search.js', reverse_lazy("javascript-catalog"))
+
+    def __init__(self, campus_id=1, max_results=5, url_field=None, attrs=None):
+        """
+        :param campus_id: Campus to search for points of interest on. Default: NTNU Gløshaugen
+        :param max_results: Maximum number of search results to return
+        :param url_field: Field to autofill with mazemap url. If None, autofill functionality is turned off
+        :param attrs: HTML attributes for the <input> element
+        """
+
+        default_attrs = {
+            "placeholder": _("Search places"),
+            "data-campusId": campus_id,
+            "data-maxResults": max_results,
+            "data-urlField": url_field,
+        }
+        required_class_attr = "prompt"
```

Comment:
This looks like a class attribute in regards to Python. Would probably be better code quality to define it in this way also. That is moving it to right beneath the `template_name` class attribute.